internal/image/ova.go

```diff
@@ -0,0 +1,111 @@
+package image
+
+import (
+	"archive/tar"
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/go-openapi/swag"
+	"github.com/openshift/assisted-image-service/pkg/isoeditor"
+)
+
+type Ova struct {
+	Id uint64
+	Ip *string
+}
+
+type Image interface {
+	Generate() (io.Reader, error)
+}
+
+func (o *Ova) Generate() (io.Reader, error) {
+	// Generate iginition
+	ignitionContent, err := o.generateIgnition()
+	if err != nil {
+		return nil, fmt.Errorf("error generating ignition: %w", err)
+	}
+
+	// Genreate TAR file
+	var buf bytes.Buffer
+	tw := tar.NewWriter(&buf)
+	reader, err := isoeditor.NewRHCOSStreamReader("rhcos-live.x86_64.iso", &isoeditor.IgnitionContent{Config: []byte(ignitionContent)}, nil, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error reading ISO file: %w", err)
+	}
+	isoData, err := io.ReadAll(reader)
```

Comment:
This isn't really streaming - it's loading the entire ISO to memory each time.

Reply:
Thanks, correct. Need to find a way to serve a stream.